dbt/include/spark/macros/utils/listagg.sql

```diff
@@ -0,0 +1,17 @@
+{% macro spark__listagg(measure, delimiter_text, order_by_clause, limit_num) -%}
+
+  {% if order_by_clause %}
+    {{ exceptions.warn("order_by_clause is not supported for listagg on Spark/Databricks") }}
+  {% endif %}
```

Comment:
[The docs](https://spark.apache.org/docs/latest/api/sql/index.html#collect_list) make it pretty clear that:

> The function is non-deterministic because the order of collected results depends on the order of the rows which may be non-deterministic after a shuffle.

The only way I've found to this requires a subquery to calculate `rank()` first, then passed into `collect_list` (with a `struct` and `array_sort` to boot, probably)